snuba/consumer.py

```diff
@@ -39,16 +37,11 @@ def process_message(self, message):
         if processed is None:
             return None
 
-        action_type, processed_message = processed
-
-        if action_type == processor.INSERT:
-            result = self.__dataset.row_from_processed_message(processed_message)
-        elif action_type == processor.REPLACE:
-            result = processed_message
-        else:
+        action_type = processed[0]
+        if action_type not in set([processor.INSERT, processor.REPLACE]):
             raise InvalidActionType("Invalid action type: {}".format(action_type))
```

Comment:
This entire method could just return on L38 if we are OK performing this assertion in flush instead.

Reply:
yeah, I like the idea of simplifying it.